doc/user/content/manage/troubleshooting.md

```diff
@@ -183,7 +183,66 @@ ORDER BY elapsed_ns DESC
 From the results of this query we can see that most of the elapsed time of the dataflow is caused by the time it takes to maintain the updates in one of the arrangements of the dataflow of the materialized view.
 
 ## Why is Materialize unresponsive or slow?
+Slow responses can be the result of a number of factors, which we can help you understand and address.
 
+### Indexes
+
+If you haven’t already, consider creating an [index](https://materialize.com/docs/sql/create-index/) for your query. The [optimizations](https://materialize.com/docs/transform-data/optimization/) page in our docs has more advanced guidance on building an efficient index. Indexes maintain a query’s up-to-date results in memory within a [cluster](https://materialize.com/docs/get-started/key-concepts#clusters), which allows future queries to return quickly.
```

Comment:
This is good, and I'd like to think about if MVs should be mentioned alongside indexes here too. This is especially true depending on the use patterns of the query and if other clusters would like to consume the query/index or not. The "When to Use Indexes and Materialized Views" blog post is great below. I think it's good as is, though, this is more of a musing.